### PR TITLE
os/arch/arm/src/amebad: SPI DMA data slicing

### PR DESCRIPTION
**Pull request by djn91**
1. zmalloc 32 byte alignment
2. spi dma data slicing

### CODE REVIEW

**Review by sunghan-chang — COMMENTED** [review]
@djn91 Could you find comments on PR #6154 ?

File: os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c [hunk 1]
```diff
@@ -136,7 +139,19 @@ u8 *_tizenrt_malloc(u32 sz)
 
 u8 *_tizenrt_zmalloc(u32 sz)
 {
-	return kmm_zalloc(sz);
+	void *pbuf = NULL;
```
Comment (abhishek-samsung) [inline os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c:142]: Doesn't this change all the existing code where _tizenrt_zmalloc is used? I think aligned malloc takes up more space. So, may be adding a new function might be good. Please consider this suggestion, its not a mandatory change.

**Comment by djn91** [comment]
> @djn91 Could you find comments on PR #6154 ?

yes i will update accordingly

File: os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c (hunk 1 above)
Comment (sunghan-chang) [reply os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c:142]: This change should be a different commit from spi dma change. Please split commits for them.

Comment (djn91) [reply os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c:147]: updated with new commit

**Review by sunghan-chang — COMMENTED** [review]
`add 32bit alignment` is difficult to know what it is.
How about `add new zalloc api with 32bit alignment`?

**Comment by djn91** [comment]
> `add 32bit alignment` is difficult to know what it is. How about `add new zalloc api with 32bit alignment`?

updated accordingly

**Comment by sunghan-chang** [comment]
@djn91 oops, path is necessary.
`board/rtl8730e: ........`

**Comment by djn91** [comment]
> @djn91 oops, path is necessary. `board/rtl8730e: ........`

sorry I missed that, I have updated.

**Comment by sunghan-chang** [comment]
@abhishek-samsung Could you verify this PR?

**Comment by abhishek-samsung** [comment]
> @abhishek-samsung Could you verify this PR?

I only have a peripheral that supports rx and I checked with it. It's working fine.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**